packages/configuration-screen/src/util.ts

```diff
@@ -0,0 +1,21 @@
+function promisify<T extends (...args: any[]) => void>(
+    fn: T
+): (...args: Parameters<T>) => Promise<ReturnType<T> extends void ? void : ReturnType<T>> {
+    return (...args: Parameters<T>): Promise<ReturnType<T> extends void ? void : ReturnType<T>> => {
+        return new Promise((resolve, reject) => {
+            function callback(err: any, result: any) {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(result);
+                }
+            }
+
+            args.push(callback);
+
+            fn.call(this, ...args);
+        });
+    };
+}
+
+export { promisify };
```

Comment:
Since we're now going to the extent of installing a bunch of other dependencies, what's one more? 😂 

https://www.npmjs.com/package/react-promise

Reply:
The promisify method still makes sense in our case, it's to avoid having to listen to the callback on each request if we don't need to listen to the error. The react-promise seems more useful if methods are already Promises and we want to receive the error + know if the data is still loading